pyproject.toml

```
@@ -129,3 +120,51 @@ D204
 D213
 D406
 D407"""
+
+[tool.ruff]
+select = [
+  "C9",
+  "E",
+  "F",
+  "I",
+  "PL",
+  "S",
+  "UP",
+  "W",
+  "YTT",
+]
+ignore = [
+  "E402",
+	"F403",
+	"PLR0402",
+	"PLR1711",
+	"PLR5501",
+  "PLR2004",
+  "PLW2901"
+]
+line-length = 120
+target-version = "py37"
+
+[tool.ruff.isort]
+  known-first-party = ["examples", "kornia", "test"]
+
+[tool.ruff.mccabe]
+max-complexity = 39
+
+[tool.ruff.pylint]
+max-args = 16
+max-branches = 45
+max-returns = 13
+max-statements = 258
+
+[tool.ruff.per-file-ignores]
+"*/__init__.py" = ["F401"]
+"docs/*" = ["S101", "S113"]
+"examples/train/object_detection/main.py" = ["E741"]
+"kornia/__init__.py" = ["I001"]
+"kornia/augmentation/container/base.py" = ["PLC1901"]
+"kornia/color/hls.py" = ["E741"]
```

this fixes the merge issue with SAM
```suggestion
"kornia/color/hls.py" = ["E741"]
"kornia/contrib/models/sam/architecture/image_encoder.py" = ["PLR0913"]
```